syntax = "proto3";

service AIProcess {
  rpc PreProcess (ProcessRequest) returns (ProcessReply) {}
  rpc PostProcess (ProcessRequest) returns (ProcessReply) {}
}

message ProcessRequest {
  string input = 1;
  repeated string processes = 2;
}

message ProcessReply {
  bool error = 2;
  string data = 3;
  string message = 4;
}